pytype/tools/analyze_project/parse_args.py

```diff
@@ -9,6 +10,20 @@
 
 _ARG_PREFIX = '--'
 
+def _auto_detect_cpus():
+  try:
+    return len(os.sched_getaffinity(0))
```

Comment:
This return the number of CPUs the current process is restricted to on Linux.

Reply:
What's wrong with os.cpu_count()?

Reply:
On Linux `len(os.sched_getaffinity(0))` returns the number of CPUs available to the current process, where as `os.cpu_count()` will return the total number of CPUs on the host system.

Reply:
answering my question - it's the number of available cores (not clear whether this includes hyperthreading cores).
More details: https://stackoverflow.com/questions/1006289/how-to-find-out-the-number-of-cpus-using-python/14840102

Reply:
> not clear whether this includes hyperthreading cores

As far as I know this includes hyperthreading cores. On my dual-core macbook `--jobs auto` will result in 4 parallel jobs.